frontend/src/features/settings/preferences/fields/ComparisonCriteriaOrder.tsx

```diff
@@ -0,0 +1,191 @@
+}: {
+  checkedCriteria: string[];
+  setCheckedCriteria: (target: string[]) => void;
+}) => {
+  const { t } = useTranslation();
+  const { criterias } = useCurrentPoll();
+
+  const handleUp = (index: number) => {
+    if (index === 0) return;
+    const tempCheckedCriteria = [...checkedCriteria];
+    const tempCriterion = tempCheckedCriteria[index];
+    tempCheckedCriteria[index] = tempCheckedCriteria[index - 1];
+    tempCheckedCriteria[index - 1] = tempCriterion;
+    setCheckedCriteria(tempCheckedCriteria);
+  };
+  const handleDown = (index: number) => {
+    if (index === checkedCriteria.length - 1) return;
+    const tempCheckedCriteria = [...checkedCriteria];
+    const tempCriterion = tempCheckedCriteria[index];
+    tempCheckedCriteria[index] = tempCheckedCriteria[index + 1];
+    tempCheckedCriteria[index + 1] = tempCriterion;
+    setCheckedCriteria(tempCheckedCriteria);
+  };
+
+  return (
+    <Grid flexDirection="row" container rowSpacing={2} columnSpacing={5}>
+      <Grid item xs={12}>
+        <Typography paragraph mb={0}>
+          <strong>{t('pollUserSettingsForm.criteriaPersonalization')}</strong>
+        </Typography>
+      </Grid>
+      <Grid item xs={12} sm={12} md={6}>
+        <Box mb={1}>
+          <Typography>{t('pollUserSettingsForm.optionalCriteria')}</Typography>
+          <Divider />
+        </Box>
+        {criterias
+          .filter((c) => !checkedCriteria.includes(c.name) && c.optional)
+          .map((criterion, index) => (
+            <OrderableCriterionRow
+              key={criterion.name}
+              criterion={criterion}
+              criteria={criterias}
+              checkedCriteria={checkedCriteria}
+              index={index}
+              handleDown={handleDown}
+              handleUp={handleUp}
+              setCheckedCriteria={setCheckedCriteria}
+            />
+          ))}
+        {checkedCriteria.length === criterias.length - 1 && (
+          <Alert severity="info">
+            <Typography>
+              {t('pollUserSettingsForm.optionalCriteriaEmpty')}
+            </Typography>
+          </Alert>
+        )}
+      </Grid>
+      <Grid item xs={12} sm={12} md={6}>
+        <Box mb={1}>
+          <Typography>{t('pollUserSettingsForm.displayedCriteria')}</Typography>
+          <Divider />
+        </Box>
+        {checkedCriteria.map((criterion, index) => (
+          <OrderableCriterionRow
+            key={criterion}
+            criterion={criterion}
+            criteria={criterias}
+            checkedCriteria={checkedCriteria}
+            index={index}
+            handleDown={handleDown}
+            handleUp={handleUp}
+            setCheckedCriteria={setCheckedCriteria}
+          />
+        ))}
+        {checkedCriteria.length === 0 && (
+          <Alert severity="info">
+            <Typography>
+              {t('pollUserSettingsForm.displayedCriteriaEmpty')}
+            </Typography>
+          </Alert>
+        )}
+      </Grid>
+    </Grid>
```

Comment:
2 small comments on this component. (I know this is still a draft, so feel free to ignore my comments :wink: ) 

1. When checking criteria, the height of this block may change and make other sections appear at the bottom of the screen. I find that disturbing for the user. Setting a `min-height` would help to prevent the layout from changing, ideally to fit all criteria just in 1 column, no more no less.

2. An additional breakpoint may be required on medium screens. Some space is currently lost on the right of the screen, and force the buttons to wrap on a second line:
![image](https://github.com/tournesol-app/tournesol/assets/4726554/afd19bec-1816-4766-9e46-fac496e092f6)